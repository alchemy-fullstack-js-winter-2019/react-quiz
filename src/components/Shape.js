import React from 'react';
import PropTypes from 'prop-types';


function Shape({ FontColor, BackgroundColor, ImageText }) {

    return (

        <div style={{ width: '100px', height: '100px', background: BackgroundColor, color: FontColor  }}>
            {ImageText}
        </div>  
    );
}

Shape.propTypes = {
    FontColor: PropTypes.string.isRequired,
    BackgroundColor: PropTypes.string.isRequired,
    ImageText: PropTypes.string.isRequired
};

export default Shape;
