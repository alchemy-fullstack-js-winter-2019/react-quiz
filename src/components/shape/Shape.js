import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import TextSelector from '../text/TextSelector';
import ColorSelector from '../color/ColorSelector';

function Shape({ text, fontColor, backgroundColor, handleChange }) {
  return (
    <Fragment>
      <TextSelector
        handleChange={handleChange}
        text={text}
      />
      <ColorSelector 
        fontColor={fontColor}
        backgroundColor={backgroundColor}
        handleChange={handleChange}
      />

      <h2>{text}</h2>
    </Fragment>
  );
}

Shape.propTypes = {
  handleChange: PropTypes.func.isRequired,
  text: PropTypes.string.isRequired,
  fontColor: PropTypes.string.isRequired,
  backgroundColor: PropTypes.string.isRequired
};

export default Shape;
