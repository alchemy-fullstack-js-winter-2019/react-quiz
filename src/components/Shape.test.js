import React from 'react';
import renderer from 'react-test-renderer';
import Shape from './Shape';


describe('Shape Test', () => {
    it('matches a snapshot', () => {
        const tree = renderer.create(
            <Shape 
                BackgroundColor={'white'}
                FontColor={'black'}
                ImageText={'This is a string'}
            />
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });
});
