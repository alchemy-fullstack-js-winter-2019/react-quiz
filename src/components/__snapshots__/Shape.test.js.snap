// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shape Test matches a snapshot 1`] = `
<div
  style={
    Object {
      "background": "white",
      "color": "black",
      "height": "100px",
      "width": "100px",
    }
  }
>
  This is a string
</div>
`;
